configs/terraform/modules/gcp-workload-identity-federation/main.tf

```diff
@@ -0,0 +1,24 @@
+resource "google_iam_workload_identity_pool" "main" {
+  project                   = var.project_id
+  workload_identity_pool_id = var.pool_id
+  disabled                  = false
+}
+
+resource "google_iam_workload_identity_pool_provider" "main" {
+  project                            = var.project_id
+  workload_identity_pool_id          = google_iam_workload_identity_pool.main.workload_identity_pool_id
+  workload_identity_pool_provider_id = var.provider_id
+  disabled                           = false
+  attribute_mapping                  = var.attribute_mapping
+
+  oidc {
+    issuer_uri = var.issuer_uri
+  }
+}
+
+resource "google_service_account_iam_member" "service_account" {
+  for_each           = var.sa_mapping
+  service_account_id = each.value.sa_name
+  role               = "roles/iam.workloadIdentityUser"
+  member             = "principalSet://iam.googleapis.com/${google_iam_workload_identity_pool.main.name}/${each.value.attribute}"
```

Comment:
I think using `principalSet` instead `principal` was the cause of the issues when we try to use a google provided module.